app/infrastructure/cache/redis_cache.py

```
@@ -0,0 +1,230 @@
+    Args:
+        key_prefix: Prefix for cache key (e.g. 'permission').
+        ttl: Time-to-live in seconds.
+        key_builder: Optional callable(args, kwargs) -> key; else built from args/kwargs.
+
+    Returns:
+        Decorator that caches return value when CacheService is in args or kwargs.
+    """
+
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(func)
+        async def wrapper(*args: Any, **kwargs: Any) -> Any:
```

Ruff (0.14.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)